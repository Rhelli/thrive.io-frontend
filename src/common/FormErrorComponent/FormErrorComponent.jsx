import React from 'react';
import PropTypes from 'prop-types';
import styles from './FormErrorComponent.module.scss';

const FormErrorComponent = ({ errorMessage }) => (
  <>
    <p className={styles.errorMessage}>{errorMessage}</p>
  </>
);

FormErrorComponent.defaultProps = {
  errorMessage: '',
};

FormErrorComponent.propTypes = {
  errorMessage: PropTypes.string,
};

export default FormErrorComponent;
