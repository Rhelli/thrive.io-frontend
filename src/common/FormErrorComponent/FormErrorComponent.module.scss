.errorMessage {
  color: red;
  font-weight: 700;
  margin: 5px 0;
}
