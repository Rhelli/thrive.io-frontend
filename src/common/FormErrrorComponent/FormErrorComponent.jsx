import React from 'react';
import PropTypes from 'prop-types';

const FormErrorComponent = ({ errorMessage }) => (
  <div>
    <p>{errorMessage}</p>
  </div>
);

FormErrorComponent.defaultProps = {
  errorMessage: '',
};

FormErrorComponent.propTypes = {
  errorMessage: PropTypes.string,
};

export default FormErrorComponent;
