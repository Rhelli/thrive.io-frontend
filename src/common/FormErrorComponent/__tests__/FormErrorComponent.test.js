import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import FormErrorComponent from '../FormErrorComponent';

const formErrorComponent = (
  <FormErrorComponent errorMessage="error!" />
);

describe('The appearance and functionality of the FormErrorComponent', () => {
  it('Matches the FormErrorComponent snapshot', () => {
    const tree = renderer.create(formErrorComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Contains a single paragraph element', () => {
    const { container } = render(formErrorComponent);
    expect(container.firstChild.nodeName).toBe('P');
    expect(container.firstChild.classList.contains('errorMessage')).toBe(true);
  });

  it('Renders the text passed into it via props', () => {
    const { getByText } = render(formErrorComponent);
    getByText('error!');
  });
});
