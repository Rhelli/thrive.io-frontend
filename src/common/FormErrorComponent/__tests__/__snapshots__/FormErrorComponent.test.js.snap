// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the FormErrorComponent Matches the FormErrorComponent snapshot 1`] = `
<p
  className="errorMessage"
>
  error!
</p>
`;
